compiler/resolution/ResolutionCandidate.cpp

```diff
@@ -649,18 +670,36 @@ bool ResolutionCandidate::checkResolveFormalsWhereClauses() {
    */
   resolveSignature(fn);
 
+  bool isInit = strcmp(fn->name, "init") == 0 && fn->isMethod();
+  bool looksLikeCopyInit = false;
+
+  if (isInit && formalIdxToActual.size() == 3) {
+    // First formal/actual is gMethodToken
+    AggregateType* base  = getActualType(this, 1);
+    AggregateType* other = getActualType(this, 2);
+    if (base != NULL && other != NULL) {
+      if (base == other) {
+        looksLikeCopyInit = true;
+      } else if (getRootInstantiation(base) == getRootInstantiation(other)) {
+        looksLikeCopyInit = true;
+      }
+    }
+  }
```

Comment:
It seems like we need a comment here indicating why copy inits get different treatment.
Also it seems likely that computing looksLikeCopyInit belongs in another function / method, but maybe it's very specific to this code path?

Reply:
Adding a comment sounds reasonable.

I think this check is very specific to this code path, but it wouldn't hurt to move it into a local utility function.